src/main/java/edu/kit/provideq/toolbox/api/SolveRouter.java

```diff
@@ -142,6 +135,146 @@ private Mono<ServerResponse> handleSolutionRouteForMetaSolver(MetaSolver<?, ?, ?
     });
   }
 
+  private void handleRouteDocumentation(MetaSolver<?, ?, ?> metaSolver, Builder ops) {
+    var problemType = metaSolver.getProblemType();
+    ops
+        .operationId(getSolveRouteForProblemType(problemType))
+        .tag(problemType.getId())
+        .description("Solves a " + problemType.getId() + " problem. To solve the problem, "
+            + "either the meta-solver will choose the best available solver,"
+            + "or a specific solver selected in the request will be used.")
+        .requestBody(requestBodyBuilder()
+                .content(getRequestContent(metaSolver))
+                .required(true))
+        .response(getResponseOk(metaSolver))
+        .response(getResponseNotFound());
+  }
+
+  private static org.springdoc.core.fn.builders.apiresponse.Builder getResponseOk(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return responseBuilder()
+            .responseCode(String.valueOf(HttpStatus.OK.value()))
+            .content(contentBuilder()
+                    .mediaType(APPLICATION_JSON_VALUE)
+                    .example(getExampleSolved(metaSolver))
+                    .example(getExampleInvalid(metaSolver))
+                    .array(arraySchemaBuilder().schema(
+                            schemaBuilder().implementation(Solution.class))));
+  }
+
+  private static org.springdoc.core.fn.builders.exampleobject.Builder getExampleSolved(
+          MetaSolver<?, ?, ?> metaSolver) {
+    return getExampleOk(
+            "Solved",
+            metaSolver,
+            solution -> {
+              solution.setSolutionData("Solution data to solve the problem");
```

Comment:
maybe rather "solution to the problem". Would it be possible to show an actual solution here to show the format of our solutions?

Reply:
Unless you do some reflection stuff to get the generic parameter of the meta solver (or a specific solver of the meta solver), and then create a new object of that type which we then pass in JSON format to the solution data, I don't think we can do this.
Or of course, you could have an example solution like we have for problems if we want to do that. That's of course more work to keep up to date but if we have a use for it that's fine! We could even integrate that into tests.

Reply:
I'm not sure if I understood you correctly. I think having example solutions matching our example problems would be a good approach and I think keeping those up to date is worth the effort. That's what you mean, right?

Reply:
Yes, that's what I meant

Reply:
Yes, I think that would make sense. We can keep INVALID and add something like ERROR. And there should probably be an equivalent to abort() for this too.

Reply:
We have two things to think about for this one, so I would maybe merge this PR first so we have something ready for tomorrow.

1. How do we associate problem and solution?
     1.1. We could have them both in the same file (which I don't really like because we now have to distinguish what is what.
     1.2. We could have another file with the suffix `_solution` and keep it convention-based.
2. Our solution formats are far from standardized currently. For example, the dead feature model returns something like this
```
The following features are dead features:
Tomato
Cheese
```

Reply:
Okay, I've created an issue for this matter: https://github.com/ProvideQ/toolbox-server/issues/49